src/slurm_cloud_bursting/suspend.py

```diff
@@ -0,0 +1,153 @@
+from retrying import retry
+
+from common.schedulers.slurm_commands import get_nodes_info, set_nodes_idle
+from common.utils import grouper
+from slurm_cloud_bursting.utils import CONFIG_FILE_PATH
+
+LOG_CONFIG_FILE = "/opt/parallelcluster/configs/slurm/parallelcluster_suspend_logging.conf"
+log = logging.getLogger(__name__)
+
+
+class SlurmSuspendConfig:
+    DEFAULT_MAX_RETRY = 5
+    DEFAULT_MAX_INSTANCES_BATCH_SIZE = 100
+
+    def __init__(self, config_file_path=None, **kwargs):
+        if config_file_path:
+            self._get_config(config_file_path)
+        else:
+            self.region = kwargs.get("region")
+            self.cluster_name = kwargs.get("cluster_name")
+            self.max_batch_size = kwargs.get("max_batch_size")
+            self.boto3_config = kwargs.get("boto3_config")
+
+    def __repr__(self):
+        attrs = ", ".join(["{key}={value}".format(key=key, value=repr(value)) for key, value in self.__dict__.items()])
+        return "{class_name}({attrs})".format(class_name=self.__class__.__name__, attrs=attrs)
+
+    def _get_config(self, config_file_path):
+        """Get resume program configuration."""
+        log.info("Reading %s", config_file_path)
+
+        config = ConfigParser()
+        try:
+            config.read_file(open(config_file_path, "r"))
+        except IOError:
+            log.error(f"Cannot read slurm cloud bursting scripts configuration file: {config_file_path}")
+            raise
+
+        self.region = config.get("slurm_cb_config", "region")
+        self.cluster_name = config.get("slurm_cb_config", "cluster_name")
+        self.max_batch_size = int(
+            config.get("slurm_cb_config", "max_batch_size", fallback=self.DEFAULT_MAX_INSTANCES_BATCH_SIZE)
+        )
+
+        # Configure boto3 to retry 5 times by default
+        self._boto3_config = {"retries": {"max_attempts": self.DEFAULT_MAX_RETRY, "mode": "standard"}}
+        proxy = config.get("slurm_cb_config", "proxy", fallback="NONE")
+        if proxy != "NONE":
+            self._boto3_config["proxies"] = {"https": proxy}
+        self.boto3_config = Config(**self._boto3_config)
+
+        log.info(self.__repr__())
+
+
+def _delete_instances(instance_ids_to_nodename, region, boto3_config, batch_size):
+    """Terminate corresponding EC2 instances."""
+    ec2_client = boto3.client("ec2", region_name=region, config=boto3_config)
+    log.info("Terminating instances %s", list(instance_ids_to_nodename.keys()))
+    for instances in grouper(instance_ids_to_nodename.keys(), batch_size):
+        try:
+            # Boto3 clients retries on connection errors only
+            # Adding extra layer of retry on all exceptions to try to terminate instances
+            retry(stop_max_attempt_number=3, wait_fixed=5000)(ec2_client.terminate_instances)(InstanceIds=instances,)
+        except ClientError as e:
+            log.error("Failed when terminating instances %s with error %s", instances, e)
```

Comment:
also in this case we probably want to exit the script with error. Shall we return an encountered_error flag and handle this in the caller?